workers/unity/Assets/Playground/Scripts/EntityInitialization/GameObjectInitializationSystem.cs

```diff
@@ -0,0 +1,120 @@
+        private EntityGameObjectCreator entityGameObjectCreator;
+        private uint currentHandle;
+        private readonly Dictionary<int, GameObject> entityGameObjectCache = new Dictionary<int, GameObject>();
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            worker = WorkerRegistry.GetWorkerForWorld(World);
+            view = worker.View;
+            origin = worker.Origin;
+            entityGameObjectCreator = new EntityGameObjectCreator(World, new Dictionary<string, GameObject>());
+        }
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < addedEntitiesData.Length; i++)
+            {
+                var prefabMapping = PrefabConfig.PrefabMappings[addedEntitiesData.PrefabNames[i].Prefab];
+                var transform = addedEntitiesData.Transforms[i];
+                var entity = addedEntitiesData.Entities[i];
+                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i].EntityId;
+
+                if (!(worker is UnityClient) && !(worker is UnityGameLogic))
+                {
+                    Debug.LogErrorFormat(Errors.UnknownWorkerType, World.Name);
+                    continue;
+                }
+
+                var prefabName = worker is UnityGameLogic
+                    ? prefabMapping.UnityGameLogic
+                    : prefabMapping.UnityClient;
+
+                var position = new Vector3(transform.Location.X, transform.Location.Y, transform.Location.Z) + origin;
+                var rotation = new UnityEngine.Quaternion(transform.Rotation.X, transform.Rotation.Y,
+                    transform.Rotation.Z, transform.Rotation.W);
+
+                var gameObject =
+                    entityGameObjectCreator.CreateEntityGameObject(entity, prefabName, position, rotation,
+                        spatialEntityId);
+                var gameObjectReference = new GameObjectReference { GameObject = gameObject };
+
+                entityGameObjectCache[entity.Index] = gameObject;
+                var gameObjectReferenceHandleComponent = new GameObjectReferenceHandle();
+
+                PostUpdateCommands.AddComponent(addedEntitiesData.Entities[i], gameObjectReferenceHandleComponent);
+                viewCommandBuffer.AddComponent(entity, gameObjectReference);
+                worker.EntityGameObjectLinker.LinkGameObjectToEntity(gameObject, entity, spatialEntityId,
+                    viewCommandBuffer);
+            }
+
+            for (var i = 0; i < removedEntitiesData.Length; i++)
+            {
+                var entityIndex = removedEntitiesData.Entities[i].Index;
+                GameObject gameObject;
+                if (!entityGameObjectCache.TryGetValue(entityIndex, out gameObject))
+                {
+                    Debug.LogErrorFormat(Errors.GameObjectNotFound, entityIndex);
```

Comment:
Just marking Debug logs. I know this one wasn't introduced in this PR but we should still get rid of it here.

Reply:
Done.